src/components/panel/panel.js

```diff
@@ -200,6 +201,34 @@ angular
  * @returns {!MdPanelAnimation} panelAnimation
  */
 
+/**
+ * @ngdoc method
+ * @name $mdPanel#newPanelGroup
+ * @description
+ * Creates a panel group and adds it to a tracked list of panel groups.
+ *
+ * @param groupName {string} Name of the group to create.
+ * @param config {!Object=} Specific configuration object that may contain the
+ *     following properties:
+ *
+ *   - `maxOpen` - `{number=}`: The maximum number of panels that are allowed
+ *     open within a defined panel group.
+ *
+ * @returns {!Object} group
```

Comment:
@bradrich You just want a more specific type than Object. Objects can have "puppies", but I'm pretty sure if I write "$mdPanel.newPanelGroup('my-group').puppies" it will break.